narwhals/utils.py

```diff
@@ -1070,3 +1073,40 @@ def check_column_exists(columns: list[str], subset: list[str] | None) -> None:
     if subset is not None and (missing := set(subset).difference(columns)):
         msg = f"Column(s) {sorted(missing)} not found in {columns}"
         raise ColumnNotFoundError(msg)
+
+
+def _parse_time_unit_and_time_zone(
+    time_unit: TimeUnit | Iterable[TimeUnit] | None,
+    time_zone: str | timezone | Iterable[str | timezone | None] | None,
+) -> tuple[set[str], set[str | None]]:
+    time_units = (
+        {"ms", "us", "ns"}
+        if time_unit is None
+        else {time_unit}
+        if isinstance(time_unit, str)
+        else set(time_unit)
+    )
+    time_zones: set[str | None] = (
+        {None}
+        if time_zone is None
+        else {str(time_zone)}
+        if isinstance(time_zone, (str, timezone))
```

Comment:
do we have any test for when it's an instance of `timezone`? can we check that `timezone.utc` and `ZoneInfo.UTC` both work?

Reply:
Well.. `ZoneInfo` does not work for polars 😱